pkg/reconciler/namespace/namespace.go

```diff
@@ -268,6 +275,15 @@ func (c *Controller) reconcileNamespace(ctx context.Context, lclusterName string
 		ns.Labels = map[string]string{}
 	}
 
+	if ns.Labels[clusterLabel] == "" && !scheduleRequirement.Matches(labels.Set(ns.Labels)) {
```

Comment:
I think it may be potentially confusing that the logic for disabling scheduling for a namespace is both here and in `ensureScheduled`. Maybe localize the change to simplify maintenance? e.g.

```golang
schedulingDisabled := !scheduleRequirement.Matches(labels.Set(ns.Labels))
if schedulingDisabled {
  return c.ensureSchedulingDisabledStatus(...)
}
```

I'm also not sure why it makes sense to check for an empty cluster assignment. Once a namespace has been labeled with `kcp.dev/schedule=false`, why should the controller consider the value of `kcp.dev/cluster` for that namespace?

Reply:
Totally agree. I've updated the logic in 00b0aa88bc13e64e2ea8914ef6ad927ce220f5e2.